templates/cli/lib/commands/push.js.twig

```diff
@@ -645,9 +679,92 @@ const createAttribute = async (databaseId, collectionId, attribute) => {
     }
 }
 
-const pushCollection = async ({ all, yes } = {}) => {
-    let response = {};
+const deleteAttribute = async (collection, attribute) => {
+    log(`Deleting attribute ${attribute.key} of ${collection.name} ( ${collection['$id']} )`);
+
+    await databasesDeleteAttribute({
+        databaseId: collection['databaseId'],
+        collectionId: collection['$id'],
+        key: attribute.key,
+        parseOutput: false
+    });
+}
+
+const deepSimilar = (remote, local, collection) => {
+    if (local === undefined) {
+        return undefined;
+    }
+
+    const key = `${chalk.yellow(local.key)} in ${collection.name} (${collection['$id']})`;
 
+    if (remote.type !== local.type) {
+        return { key, attribute: remote, reason: `type changed from ${chalk.red(remote.type)} to ${chalk.green(local.type)}` };
+    }
+
+    if (remote.array !== local.array) {
+        return { key, attribute: remote, reason: `array changed from ${chalk.red(remote.array)} to ${chalk.green(local.array)}` };
+    }
+
+    if (remote.size !== local.size) {
+        return { key, attribute: remote, reason: `size changed from ${chalk.red(remote.size)} to ${chalk.green(local.size)}` };
+    }
+
+    if (remote.relatedCollectionId !== local.relatedCollectionId) {
+        return { key, attribute: remote, reason: `relationships collection id changed from ${chalk.red(remote.relatedCollectionId)} to ${chalk.green(local.relatedCollectionId)}` };
+    }
+
+    if (remote.twoWay !== local.twoWay) {
+        return { key, attribute: remote, reason: `relationships twoWay changed from ${chalk.red(remote.twoWay)} to ${chalk.green(local.twoWay)}` };
+    }
+
+    if (remote.twoWayKey !== local.twoWayKey) {
+        return { key, attribute: remote, reason: `relationships twoWayKey changed from ${chalk.red(remote.twoWayKey)} to ${chalk.green(local.twoWayKey)}` };
+    }
+
+    return undefined;
+}
+
+const findMatch = (attribute, attributes) => attributes.find((attr) => attr.key === attribute.key);
+
+const updatedList = async (remoteAttributes, localAttributes, collection) => {
+    const deleting = remoteAttributes.filter((attribute) => !findMatch(attribute, localAttributes));
+    const changes = remoteAttributes.map((attribute) => deepSimilar(attribute, findMatch(attribute, localAttributes), collection)).filter(attribute => attribute !== undefined);
+    let changedAttributes = [];
+
+    if (changes.length > 0) {
+        log('There is a conflict in your collection deployment');
+        drawTable(changes.map((change) => {
```

Comment:
I like the direction, but I think we should take it 1 step further.

Instead of showing only changes that require attribute recreation, we should show all changes. Including new/deleted indexes/attributes.

We could have extra columns to explain how to difference will be applied. For example, new attribute isn't destructive, but editing length or deleting index are.

Then, always ask them to confirm those changes. No matter if they are destructive or not.